yt/loaders.py

```diff
@@ -97,6 +97,13 @@ def load(
     if not fn.startswith("http"):
         fn = str(lookup_on_disk_data(fn))
 
+    # import it here cause it's costly
+    from pkg_resources import iter_entry_points
```

Comment:
It should be possible to switch to `importlib.metadata`, which would not only help with the perf overhead, but is better standardised.
It is what `flake8` currently uses so I'm hoping it well covers the need here

refs
https://docs.python.org/3.11/library/importlib.metadata.html?highlight=importlib%20metadata#entry-points

https://github.com/PyCQA/flake8/blob/b87034d22490583f99149f5d241750cc8f8e3a90/src/flake8/plugins/finder.py#L35

Reply:
changes to `importlib.metadata`